rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -0,0 +1,727 @@
+        // Build prev btc tx
+        BtcTransaction prevTx = new BtcTransaction(btcRegTestParams);
+        TransactionOutput prevOut = new TransactionOutput(btcRegTestParams, prevTx, Coin.FIFTY_COINS, federation.getAddress());
+        prevTx.addOutput(prevOut);
+
+        // Build btc tx to be signed
+        BtcTransaction btcTx = new BtcTransaction(btcRegTestParams);
+        btcTx.addInput(prevOut).setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(federation));
+        TransactionOutput output = new TransactionOutput(btcRegTestParams, btcTx, Coin.COIN, new BtcECKey().toAddress(btcRegTestParams));
+        btcTx.addOutput(output);
+
+        // Save btc tx to be signed
+        final Keccak256 rskTxHash = createHash3(1);
+        provider.getRskTxsWaitingForSignatures().put(rskTxHash, btcTx);
+        provider.save();
+        track.commit();
+
+        // Setup BridgeSupport
+        BridgeEventLogger eventLogger = mock(BridgeEventLogger.class);
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        when(activations.isActive(ConsensusRule.RSKIP146)).thenReturn(rskip146Active);
+        when(activations.isActive(ConsensusRule.RSKIP326)).thenReturn(rskip326Active);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeConstantsRegtest)
+                .withProvider(provider)
+                .withRepository(track)
+                .withEventLogger(eventLogger)
+                .withBtcLockSenderProvider(new BtcLockSenderProvider())
+                .withExecutionBlock(mock(Block.class))
+                .withBtcBlockStoreFactory(null)
+                .withActivations(activations)
+                .build();
+
+        BtcECKey privateKeyToSignWith = BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0);
+
+        List derEncodedSigs;
+
+        if(signature == null) {
+            Script inputScript = btcTx.getInputs().get(0).getScriptSig();
+            List<ScriptChunk> chunks = inputScript.getChunks();
+            byte[] program = chunks.get(chunks.size() - 1).data;
+            Script redeemScript = new Script(program);
+            Sha256Hash sigHash = btcTx.hashForSignature(0, redeemScript, BtcTransaction.SigHash.ALL, false);
+            BtcECKey.ECDSASignature sig = privateKeyToSignWith.sign(sigHash);
+            derEncodedSigs = Collections.singletonList(sig.encodeToDER());
+        } else {
+            derEncodedSigs = Collections.singletonList(signature);
+        }
+
+        BtcECKey federatorPubKey = findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeyToSignWith);
+        bridgeSupport.addSignature(federatorPubKey, derEncodedSigs, rskTxHash.getBytes());
+
+        verify(eventLogger, times(wantedNumberOfInvocations)).logAddSignature(federatorPubKey, btcTx, rskTxHash.getBytes());
+    }
+
+    @Test
+    public void addSignature_calledTwiceWithSameFederator_emitEventOnlyOnce() throws Exception {
```

Comment:
Couldn't this case be covered using `test_addSignature_EventEmitted` method? You could simply add a boolean `shouldSignTwice` .

That would also allow us to test this case before and after RSKIP326, this test only covers the after activation

Reply:
I thought of it, just thought that it would have too many parameters.

Reply:
I think we are missing a test pre RSKIP326, right? It should emit the event twice in that case